### PR TITLE
Docs for no-implicit-casts and no-implicit-dynamic

### PR DESCRIPTION
**Pull request by jmesserly**
As part of fixing https://github.com/dart-lang/sdk/issues/26782, I wanted to be able to link to some documentation for the Analyzer flags.

### CODE REVIEW

**Comment by jmesserly** [comment]
@leafpetersen does this new doc section look reasonable?
https://github.com/dart-lang/dev_compiler/blob/de1a91948b8447ffddba2d6221a463d66304170c/doc/STATIC_SAFETY.md#optional-features

I'm working on the CL to try these but wanted to have something to link to.

Uhhh, now that I think on it, I should probably say experimental in there somewhere :)

File: doc/STATIC_SAFETY.md [hunk 1]
```diff
@@ -416,6 +416,105 @@ code where a field definition in a subclass shadows the field
...
+}
+
+// error: implicit supertype Iterable<dynamic>
+class C extends Iterable { /* ... */ }
+// okay
+class C extends Iterable<dynamic> { /* ... */ }
+```
+
+It is designed to prevent accidental use of `dynamic` in code that does not intend to use it. It will lead to 
+more verbose dynamic code, so it may not be advisable for your code depending on its goals.
```
Comment (leafpetersen) [inline doc/STATIC_SAFETY.md]: "It is designed" -> "This feature is designed".  The second sentence feels unnecessary to me, and maybe more apologetic than needed.  It's an optional feature, they can use it or not.

File: doc/STATIC_SAFETY.md [hunk 2]
```diff
@@ -416,6 +416,105 @@ code where a field definition in a subclass shadows the field
...
+Or pass `--no-implicit-casts` to Dart Analyzer:
+
+```
+dartanalyzer --strong --no-implicit-casts my_app.dart
+```
+
+### Disable implicit dynamic
+
+This is an optional feature of analyzer, intended primarily for use with strong mode's inference.
+It rejects implicit use of `dynamic`, requiring it to be written explicitly. For example:
```
Comment (leafpetersen) [inline doc/STATIC_SAFETY.md]: You say this below, but worth being really clear up front that this only rejects things that are still dynamic after inference.  e.g.  "It rejects implicit uses of `dynamic` that strong mode inference fails to fill in with a concrete type, ensuring that all types are either successfully inferred or explicitly written".

**Comment by leafpetersen** [comment]
lgtm.

File: doc/STATIC_SAFETY.md (hunk 1 above)
Comment (jmesserly) [inline doc/STATIC_SAFETY.md]: Done!

File: doc/STATIC_SAFETY.md (hunk 2 above)
Comment (jmesserly) [inline doc/STATIC_SAFETY.md]: great idea, done!

**Pull request merged**